tests/e2e/configurer/chain/chain.go

```diff
@@ -41,3 +55,96 @@ func New(t *testing.T, containerManager *containers.Manager, id string, initVali
 		containerManager:     containerManager,
 	}
 }
+
+// RunNode runs a node container for the given nodeIndex.
+// The node configuration must be already added to the chain config prior to calling this
+// method.
+func (c *Config) RunNode(nodeIndex int) error {
+	c.t.Logf("starting %s validator containers...", c.Id)
+
+	resource, err := c.containerManager.RunValidatorResource(c.Id, c.NodeConfigs[nodeIndex].Name, c.NodeConfigs[nodeIndex].ConfigDir)
+	if err != nil {
+		return err
+	}
+
+	hostPort := resource.GetHostPort("26657/tcp")
+	rpcClient, err := rpchttp.New("tcp://"+hostPort, "/websocket")
+	if err != nil {
+		return err
+	}
+
+	require.Eventually(
+		c.t,
+		func() bool {
+			if _, err := rpcClient.Health(context.Background()); err != nil {
+				return false
+			}
+
+			c.t.Logf("started %s node container: %s", resource.Container.Name[1:], resource.Container.ID)
+			return true
+		},
+		5*time.Minute,
+		time.Second,
+		"Osmosis node failed to produce blocks",
+	)
+
+	c.NodeConfigs[nodeIndex].rpcClient = rpcClient
+
+	if c.NodeConfigs[nodeIndex].IsValidator {
+		return c.ExtractValidatorOperatorAddress(nodeIndex)
+	}
+
+	return nil
+}
+
+// WaitUntil waits until validator with validatorIndex reaches doneCondition. Return nil
+// if reached, error otherwise.
+func (c *Config) WaitUntil(nodeIndex int, doneCondition func(syncInfo coretypes.SyncInfo) bool) error {
+	var latestBlockHeight int64
+	for i := 0; i < waitUntilrepeatMax; i++ {
+		ctx, cancel := context.WithTimeout(context.Background(), waitUntilRepeatPauseTime)
+		defer cancel()
+		status, err := c.NodeConfigs[nodeIndex].rpcClient.Status(ctx)
+		if err != nil {
+			return err
+		}
+		latestBlockHeight = status.SyncInfo.LatestBlockHeight
+		// let the node produce a few blocks
+		if !doneCondition(status.SyncInfo) {
+			time.Sleep(waitUntilRepeatPauseTime)
```

Comment:
yeah imo we could just remove this line and increase the repeat max. Probably wont shed a ton of time off but  seconds can add up.